examples/topography_spectra.jl

```diff
@@ -0,0 +1,129 @@
+using LazyArtifacts
+using ClimaAtmos.AtmosArtifacts
+import ClimaAtmos as CA
+import ClimaCore as CC
+import ClimaComms
+import ClimaParams as CP
+import ClimaCore.Quadratures
+using ClimaUtilities: SpaceVaryingInputs
+using ClimaCore.Hypsography: diffuse_surface_elevation!
+using ClimaCore: Remapping, Geometry, Operators, Spaces
+using CairoMakie
+
+import ClimaCoreSpectra: power_spectrum_2d
+
+const AA = AtmosArtifacts
+
+
+function mask(x::FT) where {FT}
+    return x * FT(x > 0)
+end
+
+"""
+    generate_spaces(; h_elem=60, diffiter=32)
+For a given number of elements `h_elem`, this function generates a 
+horizontal spectral space (ClimaCore.SpectralElementSpace2D) and
+generates an interpolated (using SpaceVaryingInputs) orography field
+on this space. The `diffuse_surface_elevation` function is called 
+to apply `diffiter` instances of Laplacian diffusion passes 
+(which rely on the canonical unsteady diffusion equation).
+Returns a `Field` containing surface orography. 
+"""
+function generate_spaces(; h_elem = 60, diffiter = 32, planet_radius = 6.378e6)
+    FT = Float32
+    cubed_sphere_mesh =
+        CA.cubed_sphere_mesh(; radius = FT(planet_radius), h_elem)
+    quad = Quadratures.GLL{4}()
+    comms_ctx = ClimaComms.context()
+    h_space = CA.make_horizontal_space(cubed_sphere_mesh, quad, comms_ctx, true)
+    Δh_scale = Spaces.node_horizontal_length_scale(h_space)
+    @assert h_space isa CC.Spaces.SpectralElementSpace2D
+    coords = CC.Fields.coordinate_field(h_space)
+    target_field = CC.Fields.zeros(h_space)
+    elev_from_file = SpaceVaryingInputs.SpaceVaryingInput(
+        AA.earth_orography_file_path(; context = comms_ctx),
+        "z",
+        h_space,
+    )
+    elev_from_file = @. mask(elev_from_file)
+    diffuse_surface_elevation!(
+        elev_from_file,
+        κ = FT(28e7 * (30 / h_elem)^2),
+        dt = FT(1),
+        maxiter = diffiter,
+    )
+    elev_from_file = @. mask(elev_from_file)
+    return elev_from_file
+end
+
+"""
+    remap_to_array(test_var, hcoords)
+Given an Array of `hcoords`, this function remaps a ClimaCore
+Field `test_var` onto `hcoords`. 
+Returns an `Array` containing the values which make up `test_var`.
+"""
+function remap_to_array(test_var, hcoords)
+    remapper = Remapping.Remapper(axes(test_var), hcoords)
+    orog = Array(Remapping.interpolate(remapper, test_var))
+end
+
+"""
+    gen_spectra(test_var)
+Given a Field `test_var`, this function first calls `remap_to_array`
+to remap this spectral element field onto Gaussian long-lat
+coordinates, and computes the spectra.
+Returns (`w_numbers`, `power_spectrum`) which contain the spherical
+wavenumbers and the spectral energy corresponding to this array
+of `w_numbers`. 
+"""
+function gen_spectra(test_var)
+    # Remap onto 1° resolution horizontal coordinates. 
+    longpts = range(-180.0, 180.0, 360)
+    latpts = range(-90.0, 90.0, 180)
+    hcoords =
```

Comment:
The fixed resolution of this grid will create problems, e.g., for high-resolution topography. Please replace by a grid that adapts to the model resolution, e.g., with a grid spacing of 360*Δh_scale/(2*pi earth_radius).